Cannot compile for HIPBLAS (whisper_mel_calc_create_cuda is undefined)

Hi,

I'm facing an issue when trying to compile with HIPBLAS on the latest commit.
Some context first:
* **OS**: Pop OS 22.04
* **ROCm**: 6.1.0.60100-82~22.04
* **GPU**: AMD RX 7900XT.

I tried compiling for HIPBLAS using `make` using `WHISPER_HIPBLAS=1 make -j8`, and when it finished I found that it apparently ignored HIPBLAS and compiled for CPU-only for some reason.

With `cmake` though, I could get it to start compiling for HIPBLAS, but with an error:
```shellsession
muaiyadh@pop-os:~/github/whisper.cpp$ CXX=hipcc CC=hipcc cmake -B build -DWHISPER_HIPBLAS=ON -DAMDGPU_TARGETS="gfx1100" -DBUILD_SHARED_LIBS=ON
-- The C compiler identification is Clang 17.0.0
-- The CXX compiler identification is Clang 17.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/hipcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/hipcc - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.34.1") 
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Performing Test HIP_CLANG_SUPPORTS_PARALLEL_JOBS
-- Performing Test HIP_CLANG_SUPPORTS_PARALLEL_JOBS - Success
-- HIP and hipBLAS found
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- x86 detected
-- Configuring done (4.1s)
-- Generating done (0.0s)
-- Build files have been written to: /home/muaiyadh/github/whisper.cpp/build
muaiyadh@pop-os:~/github/whisper.cpp$ cmake --build build -j --config Release
[  2%] Building C object CMakeFiles/whisper.dir/ggml-alloc.c.o
[  4%] Building C object CMakeFiles/whisper.dir/ggml.c.o
[  6%] Building C object CMakeFiles/whisper.dir/ggml-backend.c.o
[  8%] Building C object CMakeFiles/whisper.dir/ggml-quants.c.o
[ 10%] Building CXX object CMakeFiles/whisper.dir/ggml-cuda/arange.cu.o
[ 12%] Building CXX object CMakeFiles/whisper.dir/ggml-cuda/acc.cu.o
...
many lines. only warnings, no errors
...
[ 74%] Linking CXX shared library libwhisper.so
[ 74%] Built target whisper
[ 78%] Building CXX object examples/CMakeFiles/common.dir/common-ggml.cpp.o
[ 78%] Building CXX object examples/CMakeFiles/common.dir/common.cpp.o
[ 80%] Building CXX object examples/CMakeFiles/common.dir/grammar-parser.cpp.o
[ 82%] Building CXX object examples/bench/CMakeFiles/bench.dir/bench.cpp.o
[ 85%] Linking CXX executable ../../bin/bench
ld.lld: error: undefined reference due to --no-allow-shlib-undefined: whisper_mel_calc_create_cuda(ggml_backend*, whisper_filters const&)
>>> referenced by ../../libwhisper.so.1.6.2
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [examples/bench/CMakeFiles/bench.dir/build.make:102: bin/bench] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:313: examples/bench/CMakeFiles/bench.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 87%] Linking CXX static library libcommon.a
[ 87%] Built target common
gmake: *** [Makefile:146: all] Error 2
muaiyadh@pop-os:~/github/whisper.cpp$
```

I saw that PR #2206 recently changed/added stuff related to `whisper_mel_calc_create_cuda`.

After checking out to af5833e29819810f2d83228228a9a3077e5ccd93, I could compile with HIPBLAS, and it works great.

(Btw, thank you for your wonderful work!)

--- a/Makefile
+++ b/Makefile
@@ -297,10 +297,10 @@ ggml-cuda/%.o: ggml-cuda/%.cu ggml-cuda/%.cuh ggml.h ggml-common.h ggml-cuda/com
 
 ggml-cuda.o: ggml-cuda.cu ggml-cuda.h ggml.h ggml-backend.h ggml-backend-impl.h ggml-common.h $(wildcard ggml-cuda/*.cuh)
 	$(NVCC) $(NVCCFLAGS) $(CXXFLAGS) -Wno-pedantic -c $< -o $@
-endif
 
 whisper-mel-cuda.o: whisper-mel-cuda.cu whisper.h ggml.h ggml-backend.h whisper-mel.hpp whisper-mel-cuda.hpp
 	$(NVCC) $(NVCCFLAGS) $(CXXFLAGS) -Wno-pedantic -c $< -o $@
+endif
 
 ifdef WHISPER_HIPBLAS
 	ROCM_PATH   ?= /opt/rocm

--- a/whisper.cpp
+++ b/whisper.cpp
@@ -3167,7 +3167,7 @@ struct mel_calc_cpu : public whisper_mel_calc {
 }
 
 whisper_mel_calc * whisper_mel_calc_create(ggml_backend_t backend, const whisper_filters & filters) {
-#if GGML_USE_CUDA
+#if defined(GGML_USE_CUDA) && !defined(GGML_USE_HIPBLAS)
     if (ggml_backend_is_cuda(backend)) {
         auto ret = whisper_mel_calc_create_cuda(backend, filters);
         // run a warmup to avoid the first kernel launch overhead (thus we get the best perf even on the first run)

